Sources/GDBRemote/Structures.cpp

```diff
@@ -165,7 +166,35 @@ std::string ProcessThreadId::encode(CompatibilityMode mode) const {
   return ss.str();
 }
 
-void StopInfo::reasonToString(std::string &key, std::string &val) const {
+void StopInfo::getWatchpointInfo(std::string &key, std::string &val,
+                                 CompatibilityMode mode, bool encodeHex) const {
```

Comment:
Instead of having an `encodeHex` argument, you can deduce this information from `mode`. That reduces the number of arguments, and centralizes all the protocol-related handling in one function (instead of having some of it here, and some of it on the call site).